src/os/bluestore/BlueStore.cc

```diff
@@ -5435,9 +5447,21 @@ void BlueStore::_close_path()
   path_fd = -1;
 }
 
-int BlueStore::_write_bdev_label(CephContext *cct,
-				 const string &path, bluestore_bdev_label_t label)
-{
+void BlueStore::_unalloc_bdev_label_offset(uint64_t offset) {
+  ceph_assert(alloc);
+  bool existing_allocation_in_offset = true;
+  alloc->foreach([&](uint64_t allocation_offset, uint64_t allocation_length) {
+      if (offset >= allocation_offset && offset <= allocation_offset+allocation_length) {
+      existing_allocation_in_offset = false;
+      }
+      });
+
+  if (!existing_allocation_in_offset) {
+    alloc->init_rm_free(offset, min_alloc_size);
+  }
+}
+
+int BlueStore::_replicate_bdev_label(const string &path, bluestore_bdev_label_t label, bool alloc_available) {
```

Comment:
Why `_replicate_bdev_label` is separate from `_write_bdev_label`?
All of them should always contain the same data.
I think that the additional locations of label should be handled inside `_write_bdev_label()`.

Reply:
IMO  better then to have _write_bdev_label  being responsible specifically for writing labels to multiple destinations and to introduce another function to tag label's extents unallocated?